web/modules/custom/custom_page/src/Controller/CustomPageController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\custom_page\Controller\CustomPageController.
+ */
+
+namespace Drupal\custom_page\Controller;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\og\Og;
+use Drupal\user\Entity\User;
+
+// @todo: Fix the description.
+/**
+ * Class CustomPageController.
+ *
+ * @package Drupal\custom_page\Controller
+ */
+class CustomPageController extends ControllerBase {
+  // @todo: Fix the description.
+  /**
+   * Controller for the base form .
+   *
+   * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
+   *   The collection rdf_entity.
+   *
+   * @return array
+   *   Return the form array to be rendered.
+   */
+  public function add($rdf_entity) {
+    // @todo: Find why value is not filtered.
+    $node = $this->entityTypeManager()->getStorage('node')->create(array(
+      'type' => 'custom_page',
+      'og_group_ref' => $rdf_entity->Id()
+    ));
+
+    // @todo: Change form name to include '_form' suffix.
+    $form = $this->entityFormBuilder()->getForm($node, 'collection_custom_page');
+
+    return $form;
+  }
+
+  /**
+   * Handles accessibility to the custom page add form through collection pages.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result object.
+   */
+  public function access(){
+    $rdf_entity = \Drupal::routeMatch()->getParameter('rdf_entity');
+    $account = \Drupal::currentUser();
+
+    if($account->isAnonymous()){
+      return AccessResult::forbidden();
+    }
+
+    if($rdf_entity->bundle() != 'collection' ){
+      return AccessResult::forbidden();
+    }
+
+    // @todo: Fix the visibility to include og membership role dependency after ISAICP-2362.
+    $user = User::load($account->id());
+    if(!(Og::isMember($rdf_entity, $user))){
+      return AccessResult::forbidden();
+    }
+
+    return AccessResult::allowed();
```

Comment:
I am not 100% sure but I think that allowed is the way to go here. The discussion on the 166 was about a field level access overriding the neutral access returned by all other handlers. 
Also, node is returning a neutral access in order to be better integrated with the rest of the modules.
But in our case, we are overriding the default controller and form in order to alter it a bit and we provide our custom method to handle access through the route. In addition to that, we are not calling for any invocations nor are we part of any core flow since we overrode the access method. That makes this method the "high bidder". 
So, considering that if all elements return a neutral access, then the access is denied, the high bidder is the one who should take the responsibility to allow the access.

Also, changing this to a neutral denies access to the page (because there is no method specifically allowing access and since no method explicitly cares for the access, the system does not care that much and denies it as it should) which backs up all the above a bit. :)